src/components/forms/SubmitTemplate/server-action.ts

```diff
@@ -0,0 +1,50 @@
+'use server';
+
+import { getSession, saveTemplate } from '@utils/form-actions';
+import { redirect, RedirectType } from 'next/navigation';
+import { templateFromSessionMapper, validateTemplate } from '@domain/templates';
+import { logger } from '@utils/logger';
+
+export async function submitTemplate(formData: FormData) {
+  const sessionId = String(formData.get('sessionId'));
+
+  const session = await getSession(sessionId);
+
+  if (!session) {
+    return redirect('/invalid-session', RedirectType.push);
+  }
+
+  try {
+    if (session.templateType === 'UNKNOWN') {
```

Comment:
worth checking that this is a recognised template type rather than checking it isn't unknown? as a guard against rogue POST requests

Reply:
So this is handled in both `createTemplateFromSession` which throws an error if the templateType is unknown or not configured. 

Similarly with the zod validation. If the`type` is not configured within zod's union the validation will fail.